core/src/main/java/org/elasticsearch/search/SearchShardTarget.java

```diff
@@ -138,8 +138,8 @@ public int hashCode() {
     @Override
     public String toString() {
         if (nodeId == null) {
-            return "[_na_][" + index + "][" + shardId + "]";
+            return "[_na_]" + shardId;
         }
-        return "[" + nodeId + "][" + index + "][" + shardId + "]";
```

Comment:
This toString was all messed up because shardId has the index in it. And shardId's toString already *has* `[]` wrapping it.